conversation_reconstruction_dataflow/conversation_reconstruction/construct_utils/run_constructor.py

```diff
@@ -33,35 +34,42 @@
 from .conversation_constructor import Conversation_Constructor
 
 parser = argparse.ArgumentParser(description='Running conversation reconstruction on a list of revisions.')
-parser.add_argument('--revisions',  dest='revision_ids', required = True, help='A revision list in json format for reconstruction.')
-parser.add_argument('--table',  dest='table', required = True, help='BigQeury table where the revisions are stored.')
+parser.add_argument('--weeks',  dest='weeks', required = True, help='A list of week/year combination in json format for reconstruction.')
+parser.add_argument('--table',  dest='table', required = True, help='BigQuery table where the revisions are stored.')
+parser.add_argument('--page_id',  dest='page_id', required = True, help='The id of the page for reconstruction')
 args = parser.parse_args()
 UPDATE_RATE = 200
 
 def QueryResult2json(queryresults): 
     ret = {}
-    fields = ['sha1', 'user_id', 'format', 'user_text', 'timestamp', 'text', 'page_title',\
-    'model', 'page_namespace', 'page_id', 'rev_id', 'comment', 'user_ip', 'truncated', 'record_count', 'record_index']
+    fields = ['week', 'year', 'sha1', 'user_id', 'format', 'user_text', 'timestamp', 'text', 'page_title',\
+    'model', 'page_namespace', 'page_id', 'rev_id', 'comment', 'user_ip', 'truncated', 'records_count', 'record_index']
     for ind, val in enumerate(queryresults): 
         ret[fields[ind]] = val
     return ret
 
-def run(revision_ids, table):
+def run(weeks, table, page_id):
   page_history = []
   client = bigquery_op.Client(project='wikidetox-viz')
   processor = Conversation_Constructor()
-  for ind, rev_id in enumerate(revision_ids):
-      query = ("select * from %s where rev_id=\"%s\""%(table, rev_id))
+  for ind, week in enumerate(weeks):
+      query = ("select * from %s where week==%d and year==%d and page_id == \"%s\" order by timestamp, record_index"%(table, int(week['week']), int(week['year']), page_id))
```

Comment:
Again, better to use explicit parameters as you have a lot of arguments.

Reply:
Ah this is an outdated code, sorry.